exonum-java-binding-cryptocurrency-demo/frontend/src/plugins/blockchain.js

```diff
@@ -0,0 +1,184 @@
+      if (response.data.type !== 'committed') {
+        if (--attempt > 0) {
+          return new Promise(resolve => {
+            setTimeout(resolve, ATTEMPT_TIMEOUT)
+          }).then(makeAttempt)
+        } else {
+          throw new Error('Transaction has not been found')
+        }
+      } else {
+        return response.data
+      }
+    })
+  })()
+}
+
+module.exports = {
+  install(Vue) {
+    Vue.prototype.$blockchain = {
+      generateKeyPair() {
+        return Exonum.keyPair()
+      },
+
+      generateSeed() {
+        const buffer = nacl.randomBytes(7)
+        return bigInt.fromArray(Array.from(buffer), 256).toString()
+      },
+      
+
+      createWallet(keyPair, balance) {
+
+        let buffer = MessageHead.serialize({
+          network_id: 0,
+          protocol_version: PROTOCOL_VERSION,
+          message_id: TX_WALLET_ID,
+          service_id: SERVICE_ID,
+          payload: 0 // placeholder, real value will be inserted later
+        })
+
+        let data = {
+          ownerPublicKey: Exonum.hexadecimalToUint8Array(keyPair.publicKey),
+          initialBalance: balance
+        }
+
+        const body = CreateTransactionProtobuf.encode(data).finish();
+
+        body.forEach(element => {
+          buffer.push(element)
+        });
+
+        const signature = Exonum.sign(keyPair.secretKey, buffer)
+
+        data.ownerPublicKey = Exonum.uint8ArrayToHexadecimal(data.ownerPublicKey)
+
+        return axios.post(TX_URL, {
+         protocol_version: PROTOCOL_VERSION,
+         service_id: SERVICE_ID,
+         message_id: TX_WALLET_ID,
+         signature: signature,
+         body: data
+        })
+      },
+
+      transfer(keyPair, receiver, amountToTransfer, seed) {
+        
+        let buffer = MessageHead.serialize({
+          network_id: 0,
+          protocol_version: PROTOCOL_VERSION,
+          message_id: TX_TRANSFER_ID,
+          service_id: SERVICE_ID,
+          payload: 0 // placeholder, real value will be inserted later
+        })
+
+        const data = {
+          seed: seed,
+          senderId: Exonum.hexadecimalToUint8Array(keyPair.publicKey),
+          recipientId:  Exonum.hexadecimalToUint8Array(receiver),
+          amount: amountToTransfer
+        }
+
+        const body = CreateTransactionProtobuf.encode(data).finish();
+
+        body.forEach(element => {
+          buffer.push(element)
+        });
+
+        const signature = Exonum.sign(keyPair.secretKey, buffer)
+
+        data.senderId = Exonum.uint8ArrayToHexadecimal(data.senderId)
+        data.recipientId = Exonum.uint8ArrayToHexadecimal(data.recipientId)
+
+        return axios.post(TX_URL, {
+          protocol_version: PROTOCOL_VERSION,
+          service_id: SERVICE_ID,
+          message_id: TX_TRANSFER_ID,
+          signature: signature,
+          body: data
+        }).then(response => waitForAcceptance(keyPair.publicKey, response.data))
```

Comment:
Is it intentional that here we `waitForAcceptance`, but do not in the method above?

Reply:
@dmitry-timofeev yes, expected behavior